Modules/_math.h

```diff
@@ -23,3 +23,42 @@ _Py_log1p(double x)
 }
 
 #define m_log1p _Py_log1p
+
+/*
+   wrapper for atan2 that deals directly with special cases before
+   delegating to the platform libm for the remaining cases.  This
+   is necessary to get consistent behaviour across platforms.
+   Windows, FreeBSD and alpha Tru64 are amongst platforms that don't
+   always follow C99.  Windows screws up atan2 for inf and nan, and
+   alpha Tru64 5.1 doesn't follow C99 for atan2(0., 0.).
```

Comment:
I'm curious if Windows and FreeBSD still need these workaround in 2024. But I don't ask you to test ;-)

Reply:
> I'm curious if Windows and FreeBSD still need these workaround in 2024.

I doubt, especially FreeBSD.

As I said in the description, most functions in the cmath works with libm's atan2 function (e.g. acos).  I think it's safe to remove these helpers.  But this should be heavily tested.  Here is a draft pr with this attempt: https://github.com/python/cpython/pull/122715

Reply:
> I doubt, especially FreeBSD.

Right. When I added math.fma(), I found a bug on FreeBSD. It takes a while to get it fixed upstream, but it's being fixed: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=277783 :-)